src/main/webapp/script.js

```diff
@@ -121,6 +121,139 @@ function hyperlinkText(text, link) {
   return "<a href=" + link + ">" + text + "</a>";
 }
 
+/** Fetches recipes from the server and adds them to the DOM. */
+function loadRecipes() {
+  // rowVars used to dynamically name divs of class row, for up to 3 recipes.
+  var rowVars = {};
+  let recipeCount = 0;
+  let rowCount = 0;
+  fetch('/display-recipes').then(response => response.json()).then((recipes) => {
+    const recipeGrid = document.getElementById('recipe-grid');
+    recipes.forEach((recipe) => {
+      // Every three live streams, create a new row.
+      if (recipeCount % 3 == 0) {
+        rowCount++;
+        rowVars['recipeRow' + rowCount] = document.createElement('div');
+        rowVars['recipeRow' + rowCount].className = "row";
+      }
+      rowVars['recipeRow' + rowCount].appendChild(createFeedElement(recipe));
+      recipeGrid.appendChild(rowVars['recipeRow' + rowCount]);
+      recipeCount++;
+    })
+  });
+}
+ 
+/** Creates an element that represents a feed item,
+    for example a Recipe or Live Stream. */
+function createFeedElement(item) {
+  const feedItem = document.createElement('div');
+  feedItem.className = 'col feed-img-container';
+  // Using a constant image because, as is, recipes doesn't support photos.
+  feedItem.innerHTML += "<img src=" + "https://tinyurl.com/y8eph3n6" + ">";
+  // On click, redirect to corresponding recipe.
+  feedItem.onclick = function() {
+    window.location="/recipe.html?key=" + item.key;
+  }
+ 
+  const overlay = document.createElement('div');
+  overlay.className = "overlay";
+ 
+  const unorderedList = document.createElement('ul');
+  unorderedList.className = "list-unstyled";
+ 
+  const listElement = document.createElement('li');
+  listElement.className = "list-space";
+  listElement.innerText= item.name;
+  
+  unorderedList.appendChild(listElement);
+  overlay.appendChild(unorderedList);
+  feedItem.appendChild(overlay);
+  return feedItem;
+}
+
+function recipePageInit() {
+  getRecipeInfo();
+  loadComments();
+}
+
+function getRecipeInfo() {
+  var url = window.location.href;
+  var key = url.split('?')[1];
+
+  fetch('/new-recipe?' + key).then(response => response.json()).then(recipe => {
+    document.getElementById('recipe-title').innerHTML = recipe.name;
+    document.getElementById('recipe-description').innerHTML = recipe.description;
+    displayTags(recipe.tags);
+    displayIngredients(recipe.ingredients);
+    displaySteps(recipe.steps);
+  });
+}
+
+/** Formats and displays tags on the page. */
+function displayTags(tagsList) {
+  var tagSection = document.getElementById('recipe-tags');
+  let tagCount = 0;
+  tagsList.forEach((tag) => {
+    tagCount++;
+    tagSection.innerHTML += "#" + tag
+    if (tagCount < tagsList.length) {
+      tagSection.innerHTML += ", ";
+    }
+  });
+}
+
+/** Formats and displays ingredients, with corresponding checkboxes, on the page. */
+function displayIngredients(ingredList) {
```

Comment:
nit: you can use ingredient* instead of ingred* for variable names.

Reply:
I used ingred for all variable names because I think the abbreviation isn't easily confused with other things (but I may be wrong? Please correct me if you think of another use case) and it shortens the overall variable name (because ingredients is very long :(), so I'll leave it as is.